ApiBundle/Transformer/ContentCollectionTransformer.php

```diff
@@ -23,7 +24,9 @@ public function transform($contentCollection, $contentType = null)
         $facade = new ContentCollectionFacade();
 
         foreach ($contentCollection as $content) {
-            $facade->addContent($this->getTransformer('content')->transform($content));
+            if ($this->authorizationChecker->isGranted(ContentTypeForContentPanelStrategy::ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT, $content)) {
```

Comment:
Access role means the right to access it = see it, like in the left panel. I think it's not enough to only hide the edit button when you don't have the access role. The content should be hidden too.